### ISSUE TITLE
FinanceModel variation for Economics: USUAL and STRANGE beliefs

### ISSUE THREAD

**Issue by sbenthall** [sbenthall/SHARKFin#60]
Building on #58 ....

For the ECON version of the model, we need a new FinanceModel that reflects the following:

There are two possible belief regimes, USUAL and STRANGE.

The USUAL belief regime expects a LogNormal distribution of market returns with a pre-calibrated mean and standard deviation.

Each day of the simulation, the FinanceModel compares the weighted history with the USUAL distribution. If the recent history does not fit the USUAL distribution, all agents that belief the market is USUAL change their beliefs to STRANGE with some probability. [In other words, agents change beliefs from USUAL to STRANGE with a probability inversely correlated with the likelihood of weighted recent prices under the USUAL expectations.]

The STRANGE belief regime is also of LogNormal expectations, but it is inferred from a weighted history of market price data. [We can use John's original 4-parameters expectations process for now. We may use a simpler version, more familiar to economists, for the ECON model; let's put this in a separate ticket.]

Also, the agents with STRANGE beliefs have a probability of changing their beliefs to USUAL with probability that this is correlated with the likelihood of recent weighted price history under USUAL conditions.

**Comment by llorracc** [comment]
A further thought on all of this:  We should start with a version of the market model where whatever peculiarities can generate crazy outcomes are tamped down, so that the actual history generated by the model looks close enough to the consumers' assumptions that few consumers would conclude that their assumptions are wrong.  Once we have a setup in which consumers' "USUAL" beliefs are arguably rational (consistent with observed facts), we will have earned the right to explore the variants of the market model in which crazy stuff happens.

**Comment by sbenthall** [comment]
Yes, that makes sense.

We can implement a LogNormalMarket class, which just acts as an exogenous IID lognormal shock.

I suppose there is a small chance that even with such a market, some agents might switch to STRANGE beliefs.

**Comment by sbenthall** [comment]
Waiting on @llorracc's student for the statistical test to govern the switch.
Can implement this with an unsophisticated test at first.

It might create more variance into the STRANGE state if agents look at the history of prices on dates that Calvo fairy tapped them, as opposed to looking at recent prices.

**Comment by sbenthall** [comment]
Because the weighting of past history data with respect to the usual/strange threshold test is still unspecified, I'm starting with a prototype that compares the values from John's memory model to the usual expectations.

On issue that comes up is that sometimes the expected return according to John's model is negative, which is

**Comment by sbenthall** [comment]
@alanlujan91 to mind meld with Mateo to get a sense of the hypothesis test math.

@sbenthall to do individualized price histories based on attention and a switching mechanism that can be improved by other math.
